nautobot/extras/jobs.py

```diff
@@ -483,6 +484,15 @@ def as_form(cls, data=None, files=None, initial=None, approval_view=False):
             dryrun_default = cls.dryrun_default
             task_queues = cls.task_queues
 
+        # Initialize job_queue choices
+        form.fields["job_queue"] = DynamicModelChoiceField(
+            display_field="display_with_worker_count",
+            queryset=JobQueue.objects.all(),
```

Comment:
Shouldn't this be populated from / restricted to `job_model.job_queues` instead of all possible queues?

Reply:
The `query_params` below actually filters the result to be job queues that are assigned to the job.

Reply:
Changed it to `queryset=JobQueue.objects.filter(jobs=job_model),`